hip/test/utils.hip.hpp

```diff
@@ -40,31 +40,22 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include <ginkgo/core/base/executor.hpp>
 
 
+#include "core/test/gtest/environments.hpp"
 #include "hip/base/device.hpp"
 
 
 namespace {
 
 
-class HipEnvironment : public ::testing::Environment {
-public:
-    void TearDown() override { gko::kernels::hip::reset_device(0); }
-};
-
-testing::Environment* hip_env =
-    testing::AddGlobalTestEnvironment(new HipEnvironment);
-
-
 class HipTestFixture : public ::testing::Test {
 protected:
     HipTestFixture()
         : ref(gko::ReferenceExecutor::create()),
-#ifdef GKO_TEST_NONDEFAULT_STREAM
-          exec(gko::HipExecutor::create(
-              0, ref, false, gko::default_hip_alloc_mode, stream.get()))
-#else
-          exec(gko::HipExecutor::create(0, ref))
-#endif
+          stream(ResourceEnvironment::hip_device_id),
+          exec(gko::HipExecutor::create(ResourceEnvironment::hip_device_id, ref,
+                                        false, gko::default_hip_alloc_mode,
+                                        stream.get())),
+          guard(exec->get_scoped_device_id_guard())
```

Comment:
This should still differentiate between default and non-default stream

Reply:
Is there any documentation on that? Because I had errors when running the tests on different devices (although I don't remember the details anymore)

Reply:
I have to check again, might be that the issue was somewhere else.

Reply:
This commit has more details on the issue I had: c11df1a9

Reply:
I could not reproduce the issue, I'm checking to see now if this pops up in CI.

Reply:
Seems to be fine without the changes